src/components/SignUp.tsx

```diff
@@ -0,0 +1,95 @@
+import {Button, Card, Help, InputField, LogoIcon} from '@dhis2/ui'
+import {useCallback, useState} from 'react'
+import {postSignUp} from '../api'
+import styles from './SignUpPage.module.css'
+import {Navigate} from "react-router";
+
+const SignUpPage = () => {
+    const [username, setUsername] = useState('')
+    const [password, setPassword] = useState('')
+    const [confirmPassword, setConfirmPassword] = useState('')
+    const [isCreated, setIsCreated] = useState(false)
+    const [signUpError, setSignUpError] = useState('')
+
+    const doSignUp = useCallback(async () => {
+        if (password !== confirmPassword) {
+            setSignUpError("passwords doesn't match")
+            return
+        }
+
+        try {
+            const result = await postSignUp(username, password)
+            if (result.status === 201) {
+                setIsCreated(true)
+            }
```

Comment:
This leaves us with the app unresponsive if we were to get anything other than a 201. We should probably add an `else` which throws an error.

Reply:
How about...

            if (result.status === 201) {
                setIsCreated(true)
            } else {
                setSignUpError(result.data)
            }

Because if it's not 201, an error has quite likely occurred.

Reply:
Yeah that works too. Probably easier than throwing actually, because then the call to `setSignUpError` would be happening in the catch-block, and you'd have to either ensure the object structure matches, or that the code in the catch-block handles plain errors too.